packages/types/src/documents/app/row.ts

```diff
@@ -1,22 +1,76 @@
   LONGFORM = "longform",
+  // similar to string type, called Options within Budibase. This works very similarly to
+  // the string type within the backend, but is validated to a list of options. This will
+  // be displayed a select input within the builder/client.
   OPTIONS = "options",
+  // a primitive type, stores a number, as a floating point, called Number within Budibase.
+  // this type will always represent numbers as reals/floating point - there is no integer only
+  // type within Budibase.
   NUMBER = "number",
+  // a primitive type, stores a boolean, called Boolean within Budibase. This is often represented
+  // as a toggle or checkbox within forms/grids.
   BOOLEAN = "boolean",
+  // a JSON type, this type is always an array of strings, called Multi-select within Budibase.
+  // This type can be compared to the options type, as it functions similarly, but allows picking
+  // multiple options rather than a single option.
   ARRAY = "array",
+  // a string type, this is always a string when input/returned from the API, called Date/Time within
+  // Budibase. We utilise ISO date strings for representing dates, this type has a range of sub-types
+  // to restrict it to date only, time only and ignore timezone capabilities.
   DATETIME = "datetime",
+  // a JSON type, an array of metadata about files held in object storage, called Attachment List within
+  // Budibase. To utilise this type there is an API for uploading files to Budibase, which returns metadata
+  // that can be stored against columns of this type. Currently this is not supported on external databases.
   ATTACHMENTS = "attachment",
+  // a JSON type, similar to the attachments type, called Attachment within Budibase. This type functions
+  // much the same as the attachment list, but only holds a single attachment metadata as an object.
+  // This simpifies the binding experience of using this column type.
   ATTACHMENT_SINGLE = "attachment_single",
+  // a complex type, called Relationships within Budibase. This is the most complex type of Budibase,
+  // nothing should be stored against rows under link columns; this type simply represents the
+  // relationship between tables as part of the table schema. When rows are input to the Budibase API
+  // relationships to be made are represented as a list of row IDs to link. When rows are returned
+  // from the Budibase API it will contain a list of row IDs and display column values of the related rows.
   LINK = "link",
+  // a complex type, called Formulas within Budibase. This type has two variants, static and dynamic, with
+  // static only being supported against internal tables. Dynamic formulas calculate a provided HBS/JS binding
+  // based on the row context and enrich it when rows are being returned from the API. Static bindings calculate
+  // this when rows are being stored, so that the formula output can be searched upon within the DB.
```

Comment:
Is there not also a variant where this is either calculated on the frontend or in the backend?

Reply:
No, formulas are *always* calculated on the backend.

Reply:
ah - yeah I'm guessing its purely for the live evaluation, that is a good point - I'll add that to the type info just for reference.